### PR TITLE
Gazetteer search

### PR DESCRIPTION
**Pull request by annarieger**
This PR introduces a gazetteer search based on OSM dataset.

Many thanks to @KaiVolland for the great preliminary work!

Please review /cc @KaiVolland, @weskamm, @buehner, @marcjansen.

### CODE REVIEW

File: src/view/form/field/GazetteerCombo.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    /**
+     *
+     */
+    onBoxReady: function(){
+        var me = this;
+        me.nav = Ext.create('Ext.util.KeyNav', me.el, {
+            esc: function(){
+                me.clearValue();
+            },
```
Comment (marcjansen) [inline src/view/form/field/GazetteerCombo.js]: You shouldn't need the anonymous function:

```js
me.nav = Ext.create('Ext.util.KeyNav', me.el, {
  esc: me.clearValue,
  scope: me
});
```

File: src/view/form/field/GazetteerCombo.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                gazetteerGrid.show(combo);
+            }
+            me.doGazetteerSearch(newValue);
+        } else {
+            if(gazetteerGrid) {
+                gazetteerGrid.getEl().slideOut('t', {
+                    duration: 250,
+                    callback: function(){
+                        gazetteerGrid.hide();
+                    }
```
Comment (marcjansen) [inline src/view/form/field/GazetteerCombo.js]: The anonymous function here can probably as well be removed (if we can also provide the scope.)

File: src/view/grid/GazetteerGrid.js [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        type: 'basigx-gazetteersearch'
+    },
+
+    viewModel: {
+        data: {
+            title: 'Gazetteer',
+            hideToolTooltip: 'Gazetteer verbergen',
+            limitCboxLabel: 'Auf den sichtbaren Kartenbereich einschränken',
+            refreshBtnTooltip: 'Aktualisieren',
+            directionBtnTooltip: 'Hinweise zu Gazetteer'
```
Comment (marcjansen) [inline src/view/grid/GazetteerGrid.js]: Do we have German everywhere else?

File: src/view/grid/GazetteerGrid.js [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+                width: 400,
+                layout: 'fit',
+                bodyPadding: 5,
+                html: 'Führen Sie die Maus über die Suchergebnisse, um diese ' +
+                    'auf der Karte zu markieren.<br/> Klicken Sie auf ein ' +
+                    'Element, um die Karte darauf zu zentrieren.<br/>' +
+                    'Beachten Sie, dass einige Suchergebnisse außerhalb des ' +
+                    'sichtbaren Kartenbereichs liegen können, falls die ' +
+                    'Option "Auf den sichtbaren Kartenbereich einschränken" ' +
+                    'abgewählt ist.'
```
Comment (marcjansen) [inline src/view/grid/GazetteerGrid.js]: Should this be in a cfg or in a viewModel?

File: src/view/grid/GazetteerGrid.js [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+        // unregister listeners on grid hide
+        me.on('hide', me.unregisterListeners, me);
+    },
+
+    /**
+     *
+     */
+    onBoxReady: function(){
+        var me = this;
+        if(!me.map){
```
Comment (marcjansen) [inline src/view/grid/GazetteerGrid.js]: Shouldn't we use the `getMap()` method here? Can we be sure that `{config: {map: null}}` will be stored inside of the property `map`?

File: src/view/grid/GazetteerGrid.js [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+            directionBtnTooltip: 'Hinweise zu Gazetteer'
+        }
+    },
+
+    bind: {
+        title: '{title}'
+    },
+
+    tools:[{
+        type:'minimize',
```
Comment (marcjansen) [inline src/view/grid/GazetteerGrid.js]: missing a space

**Comment by marcjansen** [comment]
I was quite picky this time, @annarieger. I have added a couple of comments which you may want to address. Nothing to hard, though. Please merge once done. Thanks!

File: src/view/grid/GazetteerGrid.js (hunk 3 above)
Comment (annarieger) [inline src/view/grid/GazetteerGrid.js]: German tooltips and strings will be used e.g. in all buttons and other components such as [`LayerSlider`](https://github.com/terrestris/BasiGX/blob/master/src/view/container/LayerSlider.js) or [`NominatimSearch`](https://github.com/terrestris/BasiGX/blob/master/src/view/container/LayerSlider.js).
However, some components are (still?) in English (e.g. [`RedliningStyler`](https://github.com/terrestris/BasiGX/blob/master/src/view/container/RedlineStyler.js)). So there is a bit inconsistency between BasiGX classes. Probably we should discuss the language/translation concept later once again.

**Comment by annarieger** [comment]
Many thanks for the review @marcjansen! I've addressed all your remarks except the german translations - this issue should be probably discussed separately later.

Merging now..

**Pull request merged**